plugins/module_utils/prism/service_groups.py

```diff
@@ -22,6 +29,70 @@ def get_uuid(self, value, key="name", raise_error=True, no_response=False):
                     return entity["uuid"]
         return None
 
+    def _get_default_spec(self):
+        return deepcopy(
+            {
+                "name": None,
+                "service_list": [],
+            }
+        )
+
+    def _build_spec_name(self, payload, value):
+        payload["name"] = value
+        return payload, None
+
+    def _build_spec_desc(self, payload, value):
+        payload["description"] = value
+        return payload, None
+
+    def _build_spec_services(self, payload, config):
+        service_list = []
+
+        if config.get("tcp"):
+            service = {}
+            service["protocol"] = "TCP"
+            port_range_list = []
+            if "*" not in config["tcp"]:
+                for port in config["tcp"]:
+                    port = port.split("-")
+                    port_range_list.append(
+                        {"start_port": int(port[0]), "end_port": int(port[-1])}
+                    )
+            else:
+                port_range_list.append({"start_port": 0, "end_port": 65535})
+            service["tcp_port_range_list"] = port_range_list
+            service_list.append(service)
+
+        if config.get("udp"):
+            service = {}
+            service["protocol"] = "UDP"
+            port_range_list = []
+            if "*" not in config["udp"]:
```

Comment:
This logic for creating port range list can be extracted as a function and can used in both tcp and udp spec creation